File: MRIO/app/src/main/java/com/solie/mrio/MainActivity.java
package com.solie.mrio;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.content.Intent;
import android.os.Bundle;
import android.telephony.ClosedSubscriberGroupInfo;
import android.view.MenuItem;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.firebase.ui.auth.AuthUI;
import com.firebase.ui.auth.IdpResponse;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.solie.mrio.fragment.AcceptFragment;
import com.solie.mrio.fragment.ChatListFragment;
import com.solie.mrio.fragment.FindFragment;
import com.solie.mrio.fragment.SettingFragment;
import com.solie.mrio.network.NetworkCheck;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity implements OnClickChange {

    private static final int RC_SIGN_IN = 1000;

    public static Toolbar toolbar;
    public static ActionBar actionBar;
    public static ImageView toolbarImage;
    public static TextView toolbarText;

    public FrameLayout frameContainer;

    public static FirebaseUser user;

    FindFragment findFragment;
    ChatListFragment chatListFragment;
    AcceptFragment acceptFragment;
    SettingFragment settingFragment;

    public static boolean acceptOrNot;

    public static BottomNavigationView bottomNavigationView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        frameContainer = findViewById(R.id.frameContainer);
        bottomNavigationView = findViewById(R.id.bottomNav);

        toolbarImage = findViewById(R.id.toolbarImage);
        toolbarText = findViewById(R.id.toolbarText);

        findFragment = new FindFragment();
        chatListFragment = new ChatListFragment();
        acceptFragment = new AcceptFragment();
        settingFragment = new SettingFragment();


        setToolbar();
        networkCheck();

        bottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                item.setChecked(true);
                int id = item.getItemId();
                if (id == R.id.menu_find_list) {
                    getSupportFragmentManager().beginTransaction().replace(R.id.frameContainer, findFragment).commit();
                    acceptOrNot = false;
                } else if (id == R.id.menu_accept_list) {
                    getSupportFragmentManager().beginTransaction().replace(R.id.frameContainer, acceptFragment).commit();
                } else if (id == R.id.menu_chat_list) {
                    getSupportFragmentManager().beginTransaction().replace(R.id.frameContainer, chatListFragment).commit();
                } else {
                    getSupportFragmentManager().beginTransaction().replace(R.id.frameContainer, settingFragment).commit();
                }
                return true;
            }
        });
    }

    public void setToolbar() {
        toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        actionBar = getSupportActionBar();
        actionBar.setDisplayShowCustomEnabled(true);
        actionBar.setDisplayShowTitleEnabled(false);
        actionBar.setDisplayHomeAsUpEnabled(false);
    }

    private void networkCheck() {
        int status = NetworkCheck.getConnectivityStatus(getApplicationContext());
        if (status == NetworkCheck.TYPE_NOT_CONNECTED) {
            //오프라인상태
            Toast.makeText(getApplicationContext(), "인터넷 연결을 확인해주세요.", Toast.LENGTH_SHORT).show();
            finish();
        } else {
            userLoginCheck();
        }
    }

    public void userLoginCheck() {
        user = FirebaseAuth.getInstance().getCurrentUser();
        if (user == null) {
            loginProcess();
        } else {
            setFragment();
        }
    }

    private void loginProcess() {
        startActivityForResult(
                AuthUI.getInstance().createSignInIntentBuilder()
                        .setTheme(AuthUI.getDefaultTheme())
                        .setLogo(R.mipmap.ic_launcher_round)
                        .setAvailableProviders(getProviders())
                        .setTosAndPrivacyPolicyUrls("https://naver.com", "https://google.com")
                        .setIsSmartLockEnabled(false)
                        .build(), RC_SIGN_IN);
    }

    private List<AuthUI.IdpConfig> getProviders() {
        List<AuthUI.IdpConfig> providers = new ArrayList<>();
        providers.add(new AuthUI.IdpConfig.GoogleBuilder().build());
        providers.add(new AuthUI.IdpConfig.EmailBuilder().build());
        return providers;
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == RC_SIGN_IN) {
            {
                if (resultCode != 1500) {
                    IdpResponse response = IdpResponse.fromResultIntent(data);
                }
            }
        }
    }

    public void setFragment() {
        getSupportFragmentManager().beginTransaction().add(R.id.frameContainer, findFragment).commit();
    }

    public void onClickChange(int index) {
        if (index == 0) {
            getSupportFragmentManager().beginTransaction().replace(R.id.frameContainer, findFragment).addToBackStack(null).commit();
        } else {
            getSupportFragmentManager().beginTransaction().replace(R.id.frameContainer, acceptFragment).commit();
        }
    }
}

File: RC_Challenge_3/app/src/main/java/com/jcorp/rc_challenge_3/MainActivity.kt
package com.jcorp.rc_challenge_3

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.FragmentActivity
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.OnLifecycleEvent
import com.google.android.material.snackbar.Snackbar
import com.jcorp.rc_challenge_3.databinding.ActivityMainBinding

class MainActivity : FragmentActivity() {
    private lateinit var binding: ActivityMainBinding
    private val TAG = "Main"
    private val LOGIN_REQUEST = 1000
    private var isLogin = false
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d(TAG, "onCreate: onCreate() 호출")
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        Log.d(TAG, "onActivityResult: onActivityResult() 호출")
        if (requestCode == LOGIN_REQUEST) {
            isLogin = data!!.getBooleanExtra("isLogin", false)
        }
    }

    override fun onStart() {
        super.onStart()
        Log.d(TAG, "onStart: OnStart() 호출")
        if (!isLogin) {
            val intent = Intent(this, LoginIDActivity::class.java)
            startActivityForResult(intent, LOGIN_REQUEST)
        } else {
            Snackbar.make(binding.mainContainer, "로그인에 성공하였습니다.", Snackbar.LENGTH_SHORT).show()
        }
    }

    override fun onResume() {
        super.onResume()
        Log.d(TAG, "onResume: OnResume() 호출")
    }

    override fun onPause() {
        super.onPause()
        Log.d(TAG, "onPause: onPause() 호출")
    }

    override fun onStop() {
        super.onStop()
        Log.d(TAG, "onStop: onStop() 호출")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d(TAG, "onDestroy: onDestroy() 호출")
    }

    override fun onRestart() {
        super.onRestart()
        Log.d(TAG, "onRestart: onRestart() 호출")
    }
}